src/test/java/StringManipulationTest.java

```diff
@@ -19,63 +18,113 @@ public void tearDown() {
         manipulatedstring = null;
     }
 
+    @Test
+    public void testgetString(){
+        manipulatedstring.setString("Hello!");
+        String expected = "Hello!";
+        String actual = manipulatedstring.getString();
+        assertEquals(expected, actual);
+    }
     @Test
     public void testCount1() {
         manipulatedstring.setString("This is my string");
         int length = manipulatedstring.count();
         assertEquals(4, length);
     }
 
+    // Test when string has multiple punctuations.
     @Test
     public void testCount2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This isn't my string!");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);
     }
 
+    // Test when string is empty.
+    // Expected return is 0
     @Test
     public void testCount3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    // Test when string is null
+    // Expected return is 0
     @Test
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString(null);
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    // Test when string has only empty space character.
+    // Expected return is 0
+    @Test
+    public void testCount5() {
+        manipulatedstring.setString("     ");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);    }
+
+    // Test if the method removes the right characters without maintaining spacing
     @Test
     public void testRemoveNthCharacter1() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
+        String expected = "I' bttr uts0e 16tsinths trn6 rgh?";
+        String actual = manipulatedstring.removeNthCharacter(3, false);
+        assertEquals(expected, actual);
     }
 
+    // Test if the method removes the right characters while maintaining spacing
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        String expected = "I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?";
+        String actual = manipulatedstring.removeNthCharacter(3, true);
+        assertEquals(expected, actual);
     }
 
+    // Throw IllegalArgumentException if the given index is less than or equal to 0
```

Comment:
Looks like this comment is swapped with the comment on line 95

Reply:
Ups, thanks for pointing that out! Copying and pasting can be risky :)